### PR TITLE
feat: added cors plugin form

### PR DESCRIPTION
**Pull request by guoqqqi**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**
added cors plugin form 
image

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Please update this section with detailed description.

**Related issues**

fix/resolve #0001

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by juzhiyuan** [comment]
actually, the Form style needs to be updated. You could split it into 2 issues:
- [ ] Features (Logics)
- [ ] Styles & Friendly (Fields description/sample/...)

File: web/src/components/Plugin/UI/cors.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+  return (
+    <Form
+      form={form}
+      {...FORM_ITEM_LAYOUT}
+    >
+      <Form.Item
+        name="allow_origins"
+        label="allow_origins"
+        initialValue={"*"}
```
Comment (juzhiyuan) [inline web/src/components/Plugin/UI/cors.tsx:54]:
```suggestion
        initialValue="*"
```

File: web/src/components/Plugin/UI/cors.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      >
+        <Input></Input>
+      </Form.Item>
+      <Form.Item
+        name="allow_methods"
+        label="allow_methods"
+        initialValue={"*"}
+      >
+        <Input></Input>
```
Comment (juzhiyuan) [inline web/src/components/Plugin/UI/cors.tsx:63]:
```suggestion
        <Input />
```

Comment (juzhiyuan) [reply web/src/components/Plugin/UI/cors.tsx:63]: `Input` is not good here, use `Multiple Select`, please refer to Route create page.

Comment (juzhiyuan) [reply web/src/components/Plugin/UI/cors.tsx:63]: image

File: web/src/components/Plugin/UI/cors.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        tooltip="Maximum number of seconds the results can be cached.. Within this time range, the browser will reuse the last check result. -1 means no cache."
+      >
+        <InputNumber></InputNumber>
+      </Form.Item>
+      <Form.Item
+        name="allow_credential"
+        label="allow_credential"
+        valuePropName="checked"
+        initialValue={false}
+        tooltip="if you set this option to true, you can not use '*' for other options."
```
Comment (juzhiyuan) [inline web/src/components/Plugin/UI/cors.tsx:92]: i18n

File: web/src/components/Plugin/locales/en-US.ts [hunk 4]
```diff
@@ -21,4 +21,6 @@ export default {
   'component.step.select.pluginTemplate.select.option': 'Custom',
   'component.plugin.pluginTemplate.tip1': '1. When a route already have plugins field configured, the plugins in the plugin template will be merged into it.',
   'component.plugin.pluginTemplate.tip2': '2. The same plugin in the plugin template will override one in the plugins',
+  'component.pluginForm.cors.allow_credential.tooltip': 'if you set this option to true, you can not use \'*\' for other options.',
```
Comment (juzhiyuan) [inline web/src/components/Plugin/locales/en-US.ts:24]:
```suggestion
  'component.pluginForm.cors.allow_credential.tooltip': 'If you set this option to true, you can not use \'*\' for other options.',
```

File: web/src/components/Plugin/locales/en-US.ts [hunk 5]
```diff
@@ -21,6 +21,19 @@ export default {
...
+  'component.pluginForm.limit-conn.burst.tooltip': 'the number of excessive concurrent requests (or connections) allowed to be delayed.',
+  'component.pluginForm.limit-conn.default_conn_delay.tooltip': 'the latency seconds of request when concurrent requests exceeding conn but below (conn + burst).',
+  'component.pluginForm.limit-conn.key.tooltip': 'to limit the concurrency level.For example, one can use the host name (or server zone) as the key so that we limit concurrency per host name. Otherwise, we can also use the client address as the key so that we can avoid a single client from flooding our service with too many parallel connections or requests.Now accept those as key: "remote_addr"(client\'s IP), "server_addr"(server\'s IP), "X-Forwarded-For/X-Real-IP" in request header, "consumer_name"(consumer\'s username).',
+  'component.pluginForm.limit-conn.rejected_code.tooltip': 'returned when the request exceeds conn + burst will be rejected.',
+
+  'component.plugin.form': 'Form',
+  'component.plugin.format-codes.disable': 'Format JSON or YAML data',
```
Comment (juzhiyuan) [inline web/src/components/Plugin/locales/en-US.ts]: Why this PR will have those contents?

Comment (guoqqqi) [reply web/src/components/Plugin/locales/en-US.ts]: These files are conflicting files, generated after conflict resolution.

File: web/src/components/Plugin/UI/cors.tsx [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+        </Col>
+      </Row>
+    </Form.Item >
+  );
+
+  return (
+    <Form
+      form={form}
+      {...FORM_ITEM_LAYOUT}
+      initialValues={{ allow_origins_by_regex: [''] }}
```
Comment (juzhiyuan) [inline web/src/components/Plugin/UI/cors.tsx:83]: Why only add initial values for this field here?

File: web/src/components/Plugin/locales/en-US.ts [hunk 7]
```diff
@@ -21,6 +21,15 @@ export default {
...
+  'component.pluginForm.cors.allow_origins.extra': 'For example: https://somehost.com:8081',
+  'component.pluginForm.cors.allow_methods.tooltip': 'Which Method is allowed to enable CORS, such as: GET, POST etc. Multiple method use , to split. When allow_credential is false, you can use * to indicate allow all any method. You also can allow any method forcefully using ** even already enable allow_credential, but it will bring some security risks.',
+  'component.pluginForm.cors.allow_headers.tooltip': 'Which headers are allowed to set in request when access cross-origin resource. Multiple value use , to split. When allow_credential is false, you can use * to indicate allow all request headers. You also can allow any header forcefully using ** even already enable allow_credential, but it will bring some security risks.',
+  'component.pluginForm.cors.expose_headers.tooltip': '	Which headers are allowed to set in response when access cross-origin resource. Multiple value use , to split.',
```
Comment (juzhiyuan) [inline web/src/components/Plugin/locales/en-US.ts:29]: Space??

File: web/src/components/Plugin/UI/cors.tsx [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+      <Row>
+        <Col span={24}>
+          <Form.Item
+            name="allow_methods"
+            initialValue={["*"]}
+          >
+            <Select
+              mode="multiple"
+              optionLabelProp="label"
+              onChange={(value) => {
```
Comment (juzhiyuan) [inline web/src/components/Plugin/UI/cors.tsx:60]:
```suggestion
              onChange={(value: string[]) => {
```

File: web/src/components/Plugin/UI/cors.tsx [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+        name="allow_credential"
+        label="allow_credential"
+        valuePropName="checked"
+        initialValue={false}
+        tooltip={formatMessage({ id: 'component.pluginForm.cors.allow_credential.tooltip' })}
+      >
+        <Switch />
+      </Form.Item>
+
+      <Form.List name={['allow_origins_by_regex']}>
```
Comment (juzhiyuan) [inline web/src/components/Plugin/UI/cors.tsx:130]:
```suggestion
      <Form.List name="allow_origins_by_regex">
```

**Review by juzhiyuan — APPROVED** [review]

**Review by LiteSun — APPROVED** [review]

**Review by moonming — APPROVED** [review]

**Review by Yiyiyimu — APPROVED** [review]

**Pull request merged**